components/script/script_thread.rs

```diff
@@ -2017,12 +2018,13 @@ impl ScriptThread {
                     Some(browsing_context) => browsing_context.active_document(),
                     None => return warn!("Message sent to closed pipeline {}.", parent_pipeline_id),
                 };
-                let url = document.url();
+                let url = (*document.url()).clone();
                 if &url[..Position::AfterQuery] == &nurl[..Position::AfterQuery] &&
                     load_data.method == Method::Get {
                     self.check_and_scroll_fragment(fragment, parent_pipeline_id, document.r());
+                    document.set_url(nurl);
```

Comment:
We need to leave a note here saying that this is wrong, and should be fixed appropriately via #13437.